projectDocs/dev/remoteDebugging.md

```diff
@@ -0,0 +1,126 @@
+# Remote Debugging NVDA
+
+Sometimes, it is not possible to effectively debug a running copy of NVDA, especially if you rely on NVDA or another screen reader for development.
+In such cases, using a secondary machine to debug the running copy of NVDA can be helpful.
+This document explains how to use [Visual Studio Code] (VS Code) to remotely debug NVDA.
+It is recommended that you use the [Visual Studio Code Workspace Configuration for NVDA] when using VS Code as your IDE for NVDA development.
+
+Note: keyboard shortcuts used in this guide assume you are using VS Code in English with a QWERTY keyboard, and that you haven't customised your keymap.
+If they don't work, search the action in the Command Palette or Keyboard Shortcuts settings to find or change the key binding on your system.
+
+## You will need
+
+* Development machine: The machine with VS Code and all required extensions installed.
+  * It is recommended that you use VS Code with the [Remote - Tunnels] extension installed.
+  * To remotely debug NVDA, you must have the debug extension for the language you want to debug installed--[Python Debugger] for python and [C/C++] for C++.
+    The [Python C++ Debugger] is recommended for debugging Python and C++ simultaneously.
+  * Consider using the pre-configured NVDA VS Code workspace, which recommends all required extensions, and has other necessary settings configured.
+  * Alternatively, you can use Visual Studio Code for the Web to debug the remote copy of NVDA in a web browser if you wish.
+* Secondary machine: The machine with NVDA's source and the [NVDA dev environment] set up.
+  This is the machine you will use to run the copy of NVDA being debugged.
+  * As we are going to connect to VS Code, you will need to have VS Code installed.
+    Alternatively, you can use the [VS Code Standalone CLI].
+
+If you do not have (or do not want to use) two physical machines, you can use a virtual machine as the secondary machine.
```

Comment:
I have thought about this as well. However, I think that issues with the copy of NVDA running on the secondary (host) machine may interfere with the NVDA running in the development (sandbox) machine. Additionally, you'd still need to have VS Code installed in the sandbox. Not impossible, but methinks it would get very old very fast.

I've currently got a half-written PowerShell script written that automates setting up the NVDA build environment, so that creating a VM to use as the secondary machine should be relatively painless. It will still take a good long time (thanks to the astronomical size of the VS Build Tools), but theoretically it would be as simple as running the script at the (admin) CLI and just leaving it to do its thing in the background.